pdfplumber/table.py

```diff
@@ -59,16 +65,18 @@ def get_group(edge):
         else:
             return ("v", edge["x0"])
 
-    if snap_tolerance > 0:
-        edges = snap_edges(edges, snap_tolerance)
+    if snap_x_tolerance > 0 or snap_y_tolerance > 0:
+        edges = snap_edges(edges, snap_x_tolerance, snap_y_tolerance)
 
-    if join_tolerance > 0:
-        _sorted = sorted(edges, key=get_group)
-        edge_groups = itertools.groupby(_sorted, key=get_group)
-        edge_gen = (
-            join_edge_group(items, k[0], join_tolerance) for k, items in edge_groups
+    _sorted = sorted(edges, key=get_group)
```

Comment:
Should the `join_x_tolerance > 0` and `join_y_tolerance > 0` check be added here as well?

Reply:
My apologies, I could/should have been clearer about this. This is what I meant by this part of the PR message:

> This PR also changes `table.merge_edges(...)` behavior when `join_tolerance` (and `x`/`y` variants) `<= 0`, so that joining is attempted regardless, to handle cases of overlapping lines (which would still fulfill `tolerance == 0`). This is an uncommon situation but is, I believe, represents the user-expected behavior.

Essentially, `join_edge_group(...)` is still helpful, even when `join_tolerance==0`, because it combines ("joins") overlapping line segments that lay along the same infinite line.